### PR TITLE
[chore] Add metrics around the DataPoints received by the InfluxDB receiver

### PR DESCRIPTION
**Pull request by tj---**
**Description:** Receiver metrics will now be reported for the InfluxDB receiver (The implementation of reporting the metrics is similar to that of Kafka Receiver) - this includes `accepted` and `refused` data points.
The receiver {label, value} pair will be `receiver="influxdb"`

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23389

**Testing:** Did not add any new unit tests. Observed that the metrics were being collected by running the binary locally
```
# HELP cdm_otelcol_receiver_accepted_metric_points Number of metric points successfully pushed into the pipeline.
# TYPE cdm_otelcol_receiver_accepted_metric_points counter
cdm_otelcol_receiver_accepted_metric_points{instance="0.0.0.0:8888",job="prometheus_simple/0.0.0.0:8888",receiver="influxdb",service_instance_id="4e4459b0-125e-4f84-883c-80cb933d86f3",service_name="otelcontribcol",service_version="0.80.0-dev",transport="http"} 979
# HELP cdm_otelcol_receiver_refused_metric_points Number of metric points that could not be pushed into the pipeline.
# TYPE cdm_otelcol_receiver_refused_metric_points counter
cdm_otelcol_receiver_refused_metric_points{instance="0.0.0.0:8888",job="prometheus_simple/0.0.0.0:8888",receiver="influxdb",service_instance_id="4e4459b0-125e-4f84-883c-80cb933d86f3",service_name="otelcontribcol",service_version="0.80.0-dev",transport="http"} 0
```

**Documentation:** This is already documented in the [**service**](https://opentelemetry.io/docs/collector/configuration/#service) section of the collector configuration.

### CODE REVIEW

File: receiver/influxdbreceiver/receiver.go [hunk 1]
```diff
@@ -165,7 +181,9 @@ func (r *metricsReceiver) handleWrite(w http.ResponseWriter, req *http.Request)
 		}
 	}
 
-	if err := r.nextConsumer.ConsumeMetrics(req.Context(), batch.GetMetrics()); err != nil {
+	err := r.nextConsumer.ConsumeMetrics(req.Context(), batch.GetMetrics())
+	r.obsrecv.EndMetricsOp(ctx, "influxdb", batch.GetMetrics().DataPointCount(), err)
```
Comment (jacobmarble) [inline receiver/influxdbreceiver/receiver.go:185]: It might be a good idea to put `"influxdb"` in a string constant.

**Review by jacobmarble — APPROVED** [review]
Thank you!

File: receiver/influxdbreceiver/receiver.go (hunk 1 above)
Comment (tj---) [reply receiver/influxdbreceiver/receiver.go]: Done.

**Review by atoulme — APPROVED** [review]

**Comment by tj---** [comment]
Thank you for the review. Can I request this PR to be merged?

**Review by codeboten — APPROVED** [review]

**Pull request merged**